packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2,22 +2,1633 @@
+        if (suffixIcon != null)
+          end += centerLayout(suffixIcon, end);
+        if (suffix != null)
+          end += baselineLayout(suffix, end);
+        break;
+      }
+      case TextDirection.ltr: {
+        double start = left + _boxSize(icon).width;
+        double end = right;
+        if (prefixIcon != null)
+          start += centerLayout(prefixIcon, start);
+        if (label != null)
+          centerLayout(label, start);
+        if (prefix != null)
+          start += baselineLayout(prefix, start);
+        if (input != null)
+          baselineLayout(input, start);
+        if (hint != null)
+          baselineLayout(hint, start);
+        if (suffixIcon != null)
+          end -= centerLayout(suffixIcon, end - suffixIcon.size.width);
+        if (suffix != null)
+          end -= baselineLayout(suffix, end - suffix.size.width);
+        break;
+      }
+    }
+
+    if (helperError != null || counter != null) {
+      height = layout.subtextHeight;
+      baseline = layout.subtextBaseline;
+
+      switch (textDirection) {
+        case TextDirection.rtl:
+          if (helperError != null)
+            baselineLayout(helperError, right - helperError.size.width - _boxSize(icon).width);
+          if (counter != null)
+            baselineLayout(counter, left);
+          break;
+        case TextDirection.ltr:
+          if (helperError != null)
+            baselineLayout(helperError, left + _boxSize(icon).width);
+          if (counter != null)
+            baselineLayout(counter, right - counter.size.width);
+          break;
+      }
+    }
+
+    if (label != null) {
+      decoration.borderGap.start = textDirection == TextDirection.rtl
+        ? _boxParentData(label).offset.dx + label.size.width
+        : _boxParentData(label).offset.dx;
+      decoration.borderGap.extent = label.size.width * 0.75;
+    } else {
+      decoration.borderGap.start = null;
+      decoration.borderGap.extent = 0.0;
+    }
+
+    size = constraints.constrain(new Size(overallWidth, overallHeight));
+    assert(size.width == constraints.constrainWidth(overallWidth));
+    assert(size.height == constraints.constrainHeight(overallHeight));
+  }
+
+  void _paintLabel(PaintingContext context, Offset offset) {
+    context.paintChild(label, offset);
+  }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    void doPaint(RenderBox child) {
+      if (child != null)
+        context.paintChild(child, _boxParentData(child).offset + offset);
+    }
+    doPaint(container);
+
+    if (label != null) {
+      final Offset labelOffset = _boxParentData(label).offset;
+      final double labelHeight = label.size.height;
+      final double t = decoration.floatingLabelProgress;
+      // The center of the outline border label ends up a little below the
+      // center of the top border line.
+      final double floatingY = decoration.border.isOutline ? -labelHeight * 0.25 : contentPadding.top;
```

Comment:
this line throws an exception if border: null and a labelText is used

Reply:
jup, can confirm.
https://gist.github.com/BenSower/e4bda931033a0fa306722e3da3b02fc9

Reply:
Sorry to have missed this. I've filed and issue and will fix:
https://github.com/flutter/flutter/issues/14165